testplan/runnable/interactive/base.py

```diff
@@ -8,132 +8,276 @@
 from builtins import super
 from builtins import next
 from future import standard_library
+
 standard_library.install_aliases()
 import functools
 import re
 import six
 import numbers
 import threading
+import multiprocessing.dummy
+import itertools
 
-from testplan.common.config import ConfigOption
-from testplan.common.entity import (RunnableIHandler,
-                                    RunnableIHandlerConfig,
-                                    ResourceStatus)
+from testplan.common import entity
+from testplan.common import config
+from testplan.common.utils import networking
 import testplan.report
-from testplan.runnable.interactive.http import TestRunnerHTTPHandler
+from testplan.runnable.interactive import http
 from testplan.runners.base import Executor
 from testplan.runnable.interactive.reloader import ModuleReloader
 from testplan.runnable.interactive.resource_loader import ResourceLoader
 
 
-class TestRunnerIHandlerConfig(RunnableIHandlerConfig):
+class TestRunnerIHandlerConfig(config.Config):
     """
     Configuration object for
     :py:class:`~testplan.runnable.interactive.base.TestRunnerIHandler` runnable
     interactive handler.
     """
+
     @classmethod
     def get_options(cls):
-
-        return {ConfigOption('http_handler',
-                             default=TestRunnerHTTPHandler): object}
+        return {"target": object, "startup_timeout": int, "http_port": int}
```

Comment:
I'm not sure about not using the ConfigOption wrapper, but specify the default value in the constructor. For one, this introduce inconsistency in our code base. Secondly, I remember I fixed some occurrence of such but I cannot remember what exactly is not working.

Reply:
This is in line with our recent change in direction to add explicit named parameters to class constructors instead of relying on **kwargs. If we are specifying the default value already in the `__init__()` signature, it doesn't make sense to also define a default using ConfigOption.